src/domain/model/test/Results.ts

```diff
@@ -0,0 +1,9 @@
+import { UserId } from '../user/UserId';
+
+export class Results {
+  constructor(
+    public createdBy: UserId,
+    readonly creationTime: Date = new Date(),
+    public details: object
```

Comment:
I would rearrange the arguments to the constructor.  You'll rarely set the creationTime, but you may well use it as
```
new Results(userId, results).
```